js/classes/entities/midground/weapon/sword.js

```diff
@@ -1,3 +1,201 @@
+/**
+ * A class that represents Chad's sword.
+ * 
+ * @author Trae Claar
+ * @author Nathan Hinthorne
+ */
 class Sword {
+    /**
+     * Constructor for a Sword which takes an initial type.
+     * 
+     * @param {number} type the type of the Sword (please use Sword.TYPE_1, .TYPE_2, etc.)
+     * @throws {Error} Will throw an error if type is not a Sword member type.
+     */
+    constructor(type) {
+        Sword.checkType(type);
 
+        this.pos = CHAD.pos;
+        this.isAttacking = false;
+        this.hasHit = false;
+        this.speed = 0;
+        this.offsetX = 0;
+        this.offsetDirX = 0;
+
+        this.setType(type);
+
+        this.loadAnimations();
+    };
+
+    /** The size of the Sword on the spritesheet. */
+    static get SIZE() {
```

Comment:
Nit: This deviates from naming conventions that I often see in JavaScript. Typically I use `lowerCamelCase` as seen [here](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#object_classes).

If the rest of the codebase follows this naming convention, then no need to change it – just wanted to call this out.

Reply:
Thanks for the comment! I'm not sure what the usual JavaScript naming convention for static getters is, so I don't doubt that all-caps names like this are not standard. However, as you mention, this is a naming convention we decided on together early on for our codebase.